src/components/image-editor.js

```diff
@@ -0,0 +1,178 @@
+import PropTypes from 'prop-types';
+import { Button, Tooltip } from 'antd';
+import { useTranslation } from 'react-i18next';
+import DimensionsProvider from './dimensions-provider.js';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  ColumnHeightOutlined,
+  ColumnWidthOutlined,
+  RotateLeftOutlined,
+  RotateRightOutlined,
+  SyncOutlined,
+  ZoomInOutlined,
+  ZoomOutOutlined
+} from '@ant-design/icons';
+
+const DEFAULT_CROPPER_OPTIONS = {
+  viewMode: 1,
+  autoCropArea: 1
+};
+
+function ImageEditor({ file, editorRef, onCrop }) {
+  const { t } = useTranslation('imageEditor');
+  const onCropRef = useRef(null);
+  const cropperImageRef = useRef(null);
+  const cropperInstanceRef = useRef(null);
+  const [cropperModule, setCropperModule] = useState(null);
+
+  editorRef.current = {
+    getCroppedFile: async (maxWidth = Infinity, quality = 0.5) => {
```

Comment:
I guess here you could also use `IMAGE_OPTIMIZATION_QUALITY` as the default.

Reply:
Good idea, done.